src/_pytest/compat.py

```diff
@@ -162,7 +162,7 @@ def getfuncargnames(
     # it's passed as an unbound method or function, remove the first
     # parameter name.
     if is_method or (
-        cls and not isinstance(cls.__dict__.get(name, None), staticmethod)
+        cls and not isinstance(inspect.getattr_static(cls, name), staticmethod)
```

Comment:
To keep the same behavior as before, should do

```suggestion
        cls and not isinstance(inspect.getattr_static(cls, name, None), staticmethod)
```

unless you find it's not necessary.

Reply:
let's be explicit - fixed in 0bf451b

Reply:
actually its unnecessary, as far as  i understood the method existence is already known, the difference is that  now its looked up in all of the mro